test/lifecycle_test.dart

```diff
@@ -807,7 +876,10 @@ void sharedLifecycleTests<T extends react.Component>({
         // Check against the JS component to ensure no regressions.
         expect(component.state['counter'], 3);
         expect(component.state['counter'], getLatestJSCounter());
-        expect(component.lifecycleCalls, orderedEquals(getUpdatingSetStateLifeCycleCalls()));
+
+        if (!isComponent2) {
+          expect(component.lifecycleCalls, orderedEquals(getUpdatingSetStateLifeCycleCalls()));
```

Comment:
Why isn't this run in for Component2?

Reply:
I made this only for `Component` because `getUpdatingSetStateLifeCycleCalls` referenced `ReactSetStateTestComponent`, which at the moment extends from `Component`. Should I make another StateTestComponent for Component2 to perform the same test by with the newer lifecycle methods?

Reply:
Ah, gotcha. Yeah, I think we should have that, but I'd be fine splitting that out into a separate ticket as opposed to doing it in this PR. Whichever you prefer!